### PR TITLE
feat(radio-button-list) UX & type logic

### PR DESCRIPTION
**Pull request by VladislavRiabko**
- add RadioButtonList 
- add types of RadioButtonList

### CODE REVIEW

File: app/components/RadioButtonList/RadioButtonList.data.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export interface IRadioButtonItem {
+  id: string;
+  title: string;
+  isActive: boolean;
+}
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.data.ts]: refactor

File: app/components/RadioButtonList/RadioButtonList.styles.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { StyleSheet, ViewStyle } from 'react-native';
+
+export interface IRadioButtonStyles {
+  container: ViewStyle;
+}
+
+export const styles = StyleSheet.create<IRadioButtonStyles>({
+  container: {
+    flex: 1,
+  },
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.styles.ts]: use `generalStyles.flex`

File: app/components/RadioButtonList/RadioButtonList.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import _ from 'lodash';
+import React, { useCallback, useState } from 'react';
+import { FlatList, View } from 'react-native';
+
+import { RadioButton } from '../RadioButton/RadioButton';
+import { IRadioButtonItem, RadioButtonListType } from './RadioButtonList.data';
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.tsx]: `IRadioButtonListItem`

File: app/components/RadioButtonList/RadioButtonList.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import { IRadioButtonListProps } from './RadioButtonList.props';
+import { styles } from './RadioButtonList.styles';
+
+export const RadioButtonList: React.FC<IRadioButtonListProps> = ({
+  data,
+  type,
+}) => {
+  const [selectedItem, setSelectedItem] = useState<string>();
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.tsx]: remove

File: app/components/RadioButtonList/RadioButtonList.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  const [selectedItems, setSelectedItems] = useState<string[]>([]);
+
+  const renderItem = useCallback(
+    ({ item }: { item: IRadioButtonItem }) => {
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.tsx]: `IRadioButtonListItem`

File: app/components/RadioButtonList/RadioButtonList.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      const onPressSelectedItem = () => {
+        setSelectedItem(item.title);
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.tsx]: rename `onSelectedItemPress` or `onSelectedRadioButtonPress`

File: app/components/RadioButtonList/RadioButtonList.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      };
+      const onPressSelectedItems = () => {
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.tsx]: the same

File: app/components/RadioButtonList/RadioButtonList.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        const deleteType = _.find<string>(
+          selectedItems,
+          deleteItem => deleteItem === item.title,
+        );
+        if (deleteType) {
+          setSelectedItems(_.without<string>(selectedItems, deleteType));
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.tsx]: remove `<string>`

File: app/components/RadioButtonList/RadioButtonList.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  const [selectedItem, setSelectedItem] = useState<string>();
+  const [selectedItems, setSelectedItems] = useState<string[]>([]);
+
+  const renderItem = useCallback(
+    ({ item }: { item: IRadioButtonItem }) => {
+      const onPressSelectedItem = () => {
+        setSelectedItem(item.title);
+      };
+      const onPressSelectedItems = () => {
+        const deleteType = _.find<string>(
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.tsx]: remove `<string>`

File: app/components/RadioButtonList/RadioButtonList.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+          title={item.title}
+          isActive={
+            type === RadioButtonListType.Single
+              ? selectedItem === item.title
+              : !!_.find(selectedItems, findItem => findItem === item.title)
+          }
+          onPress={
+            type === RadioButtonListType.Single
+              ? onPressSelectedItem
+              : onPressSelectedItems
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.tsx]: refactor

File: app/components/RadioButtonList/RadioButtonList.tsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+          }
+        />
+      );
+    },
+    [setSelectedItem, selectedItem, selectedItems, setSelectedItems, type],
+  );
+
+  return (
+    <View style={styles.container}>
+      <FlatList data={data} renderItem={renderItem} />
```
Comment (emberglaz) [inline app/components/RadioButtonList/RadioButtonList.tsx]: where keyExtractor ?

**Review by emberglaz — CHANGES_REQUESTED** [review]

**Review by emberglaz — APPROVED** [review]

**Review by Reykjavik151 — APPROVED** [review]

**Pull request merged**